x-pack/plugins/screenshotting/README.md

```diff
@@ -3,18 +3,153 @@
 This plugin provides functionality to take screenshots of the Kibana pages.
 It uses Chromium and Puppeteer underneath to run the browser in headless mode.
 
-## API
+## Capabilities
+- Canvas workpads screenshots.
+- Dashboards screenshots.
+- Expressions screenshots.
+- PDF generation.
+- Batch screenshotting.
 
-The plugin exposes most of the functionality in the start contract.
-The Chromium download and setup is happening during the setup stage.
+## Usage
 
-To learn more about the public API, please use automatically generated API reference or generated TypeDoc comments.
+### Getting started
+After listing the `screenshotting` plugin in your dependencies, the plugin will be intitalized on the setup stage.
+The intitalization process downloads (if it is not already present) and verifies the Chromium build.
 
-## Testing Chromium downloads
+The start contract exposes a public API to interact with the plugin.
+Apart from the actual screenshotting functionality, it also provides a way for self-diagnostics.
 
-To download all Chromium browsers for all platforms and architectures:
+Here is an example of how you can take a screenshot of a Kibana URL.
+
+```typescript
+import type { CoreSetup, Plugin } from 'src/core/server';
+import type { ScreenshottingStart } from 'x-pack/plugins/screenshotting/server';
+
+
+interface StartDeps {
+  screenshotting: ScreenshottingStart;
+}
+
+class ExamplePlugin implements Plugin<void, void, void, StartDeps> {
+  setup({ http, getStartServices }: CoreSetup<StartDeps>) {
+    const router = http.createRouter();
+
+    router.get(
+      {
+        path: '/api/capture',
+        validate: {
+          query: schema.object({
+            id: schema.string(),
+          }),
+        },
+      },
+      async (context, request, response) => {
+        const [, { screenshotting }] = await getStartServices();
+        const { metrics, results } = await screenshotting
+          .getScreenshots({
+            request,
+            urls: [`http://localhost/app/canvas#/workpad/workpad-${request.query.id}`],
+          })
+          .toPromise();
```

Comment:
Nit: not sure if this should use `Rx.firstValueFrom` instead of `.toPromise()`. I think it should go with the former, in alignment with upgrading to RxJS 7: https://github.com/elastic/kibana/pull/129087/files#diff-0154cfbb823111800ea5acf098c44670bcf6035935df851983db9f2c55bfe16c